src/js/msp/MSPHelper.js

```diff
@@ -1913,19 +1944,21 @@ MspHelper.prototype.sendModeRanges = function(onCompleteCallback) {
     function send_next_mode_range() {
 
         var modeRange = MODE_RANGES[modeRangeIndex];
+        var modeRangeExtra = MODE_RANGES_EXTRA[modeRangeIndex];
```

Comment:
This is a bit confusing here - you are sending `permanentId` as as part of `MSP_MODE_RANGE_EXTRA`, but then you are not using it here to match entries for the two arrays.

I think the match by index that you are doing is a correct way of doing it, but there should be a remark in `msp.c` in betaflight/betaflight#7543 that stipulates that the elements of `MSP_MODE_RANGE` and `MSP_MODE_RANGE_EXTRA` be aligned.

Reply:
I understand the confusion. I'll add the comment in `msp.c` in [betaflight/betaflight#7543](https://github.com/betaflight/betaflight/pull/7543).